include/simdjson/inline/document.h

```diff
@@ -736,8 +736,24 @@ inline key_value_pair::key_value_pair(const std::string_view &_key, element _val
 really_inline element::element() noexcept : internal::tape_ref() {}
 really_inline element::element(const document *_doc, size_t _json_index) noexcept : internal::tape_ref(_doc, _json_index) { }
 
+namespace {
+// fastmod computes (a % 14) quickly
+// uses algorithm from https://arxiv.org/abs/1902.01961
+//  Faster Remainder by Direct Computation
+// 	Software: Practice and Experience 49 (6), 2019
+really_inline uint64_t fastmod14(unsigned int a) {
+  constexpr uint64_t M = 0xFFFFFFFF / 14 + 1;
+  uint64_t lowbits = (M  * a) & 0xFFFFFFFF;
+  return (lowbits * 14) >> 32;
+}
+}
+
 inline element_type element::type() const noexcept {
-  switch (tape_ref_type()) {
+  /* The element_type values are chosen so that x -> x % 14 does the correct mapping. We assume
+  that the types are correct, but this is no worse than before where an abort would follow. */
+  return static_cast<element_type>(fastmod14(static_cast<unsigned int>(tape_ref_type())));
+  // this is likely a hot function, so we don't want a switch case?
```

Comment:
We'll obviously want to remove this, but I imagine you had that planned already :)

Reply:
Yes. This was for review purposes.

Reply:
Removed.